packages/core/src/application.ts

```diff
@@ -133,6 +139,9 @@ export class Application extends Context {
   private _setupOperation(verb: string, path: string, spec: OperationObject) {
     const handler = spec['x-operation'];
     if (typeof handler === 'function') {
+      spec = Object.assign({}, spec);
```

Comment:
Shallow copy is fine here. My concern lies more with the `getApiSpec()` method declaring that it's performing a deep clone, when it's not.